src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1129,12 +1130,12 @@ static BluezConnection * BluezCharacteristicGetBluezConnection(BluezGattCharacte
         g_variant_dict_init(&options, aOptions);
 
         v = g_variant_dict_lookup_value(&options, "device", G_VARIANT_TYPE_OBJECT_PATH);
+        VerifyOrExit(v != nullptr, ChipLogError(DeviceLayer, "FAIL: No device option in dictionary (%s)", __func__));
 
-        VerifyOrExit(v != NULL, ChipLogError(DeviceLayer, "FAIL: No device option in dictionary (%s)", __func__));
-
-        path = g_variant_get_string(v, NULL);
+        // Stack-allocated dict still needs to be cleared after initialization
+        g_variant_dict_clear(&options);
 
-        retval = (BluezConnection *) g_hash_table_lookup(apEndpoint->mpConnMap, path);
+        retval = static_cast<BluezConnection *>(g_hash_table_lookup(apEndpoint->mpConnMap, g_variant_get_string(v, nullptr)));
```

Comment:
So then, when this map is destroyed, we will destroy the string, but is it equivalent to destroying the variant that the string is owned by? I'm not an expert on glib, but couldn't find such information anywhere, so maybe you should duplicate the string and release the variant here?

Reply:
It's only a hash table lookup call, the string returned by `g_variant_get_string` is owned by the variant all the time.

Anyway, according to the `g_variant_dict_lookup_value` the returned value is owned by the caller and should be freed... It's strange that valgrind did not report a leak here...

Reply:
Indeed, variant returned by the `g_variant_dict_lookup_value` needs to be freed, but the output from the valgrind shows some glib internals only. Now, leaks are down to:

Before:
```
==1426014==    definitely lost: 2,624 bytes in 26 blocks
==1426014==    indirectly lost: 26,964 bytes in 628 blocks
```

After:
```
==1722821==    definitely lost: 2,496 bytes in 24 blocks
==1722821==    indirectly lost: 26,880 bytes in 624 blocks
```